composer/trainer/checkpoint.py

```diff
@@ -355,20 +387,16 @@ def __init__(self, save_folder: str, interval: Union[Time, str], compression: st
         self.checkpoint_folder = os.path.join(run_directory.get_run_directory(), save_folder)
         os.makedirs(self.checkpoint_folder, mode=0o775, exist_ok=True)
         self.save_interval = interval
-        if compression == "":
-            self.write_mode = "w"
-            self.file_extension = ".tar"
-        elif compression == "gzip":
-            self.write_mode = "w:gz"
-            self.file_extension = ".tar.gz"
-        elif compression == "bzip2":
-            self.write_mode = "w:bz2"
-            self.file_extension = ".tar.bz2"
-        elif compression == "lzma":
-            self.write_mode = "w:xz"
-            self.file_extension = ".tar.lzma"
-        else:
-            raise ValueError(f"Unknown encryption mode: {compression}")
+        self.file_extension, self.write_mode = _format_from_compression(compression=compression)
+        self._save_paths = []
```

Comment:
I think it'd be fine if this list was public; then no need for the `@property`. Just make sure to have the docstring in the `attributes` secition.

Reply:
Can you also keep track of the `state.timer.get_timestamp()` at which `save_checkpoint` was called?